### PR TITLE
[DOCS] Document `bytes` and `time` params in cat API docs

### LINKED ISSUE #30202: Missing params in rest-api-specs
There are some missing params in the [rest-api-specs](https://github.com/elastic/elasticsearch/tree/master/rest-api-spec/src/main/resources/rest-api-spec/api)

cat.shards.json is missing: time
cat.indices.json is missing: time
cat.nodes.json is missing: bytes and time

### PR DESCRIPTION
**Pull request by jrodewig**
### Changes

- Corrects the request convention for several cat APIs
- Adds the `bytes` and `time` parameters to cat API docs and JSON specs

Closes #30202

### CODE REVIEW

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/2

File: docs/reference/rest-api/common-parms.asciidoc [hunk 1]
```diff
@@ -433,11 +433,6 @@ such as `1264`.
 A value of `-1` indicates {es} was unable to compute this number.
 end::memory[]
 
-tag::name[]
```
Comment (jrodewig) [inline docs/reference/rest-api/common-parms.asciidoc]: Removed with https://github.com/elastic/elasticsearch/pull/47672/files#diff-38f70f3453c23ad1fda645b01d00773cL19

**Comment by jakelandis** [comment]
While the parameters are supported for the APIs, they may not influence the result. I think we should add the documentation to the `_cat/` endpoints that are influence by the parameter. 

For example `GET  _cat/aliases?help` shows that the following is supported:
```
alias          | a                | alias name           
index          | i,idx            | index alias points to
filter         | f,fi             | filter               
routing.index  | ri,routingIndex  | index routing        
routing.search | rs,routingSearch | search routing   
```
These columns are not `bytes` or `time` and thus should not document or spec-api declare it's support. It won't error but that feels like an implementation detail/bug with the _cat APIs. 

Could you update only those endpoints that will be influenced by the parameters ?  I have included a full list below. 

```
GET  _cat/aliases?help
GET _cat/allocation?help
GET _cat/count?help
GET _cat/fielddata?help
GET _cat/health?help
GET _cat/indices?help
GET _cat/master?help
GET _cat/nodeattrs?help
GET _cat/nodes?help
GET _cat/pending_tasks?help
GET _cat/plugins?help
GET _cat/recovery?help
GET _cat/repositories?help
GET _cat/segments?help
GET _cat/shards?help
GET _cat/snapshots?help
GET _cat/tasks?help
GET _cat/templates?help
GET _cat/thread_pool?help
```

**Comment by jakelandis** [comment]
I logged https://github.com/elastic/elasticsearch/issues/47755 to get better validation around these APIs.

**Comment by jrodewig** [comment]
Thanks for the feedback @jakelandis.

I scaled back the changes with 1904efca4146f6a65e7fd2c7f2f0e3f6440fbf24.
The `bytes` and `time` parameters are now documented only when it impacts the response.

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/cat/indices.asciidoc [hunk 2]
```diff
@@ -77,6 +79,8 @@ primary shards. Defaults to `false`.
 
 include::{docdir}/rest-api/common-parms.asciidoc[tag=cat-s]
 
+include::{docdir}/rest-api/common-parms.asciidoc[tag=time]
+
```
Comment (jakelandis) [inline docs/reference/cat/indices.asciidoc]: I think this one should include bytes too
```
GET _cat/indices?h=ftt,ss&bytes=g&time=ms
```

File: docs/reference/cat/shards.asciidoc [hunk 3]
```diff
@@ -269,6 +271,8 @@ include::{docdir}/rest-api/common-parms.asciidoc[tag=master-timeout]
 
 include::{docdir}/rest-api/common-parms.asciidoc[tag=cat-s]
 
+include::{docdir}/rest-api/common-parms.asciidoc[tag=time]
+
```
Comment (jakelandis) [inline docs/reference/cat/shards.asciidoc]: i think this should also include bytes 
```
GET _cat/shards?h=sm&bytes=b
```

**Review by jakelandis — APPROVED** [review]
LGTM ~(pending the two minor changes)~

Thanks for doing this!

File: docs/reference/cat/indices.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/cat/indices.asciidoc]: It looks like `bytes` is already documented for this API:
https://github.com/elastic/elasticsearch/blame/be233f3d235b39d9315fc79974e9b69f20b1fe38/docs/reference/cat/indices.asciidoc#L48

File: docs/reference/cat/shards.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/cat/shards.asciidoc]: It looks like `bytes` is already documented for this API:
https://github.com/elastic/elasticsearch/blame/be233f3d235b39d9315fc79974e9b69f20b1fe38/docs/reference/cat/shards.asciidoc#L29

**Pull request merged**